File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/ByteArrayRequestBody.java
package com.github.kongpf8848.rxhttp;

import androidx.annotation.Nullable;

import java.io.IOException;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import okio.BufferedSink;

public class ByteArrayRequestBody extends RequestBody {

    private MediaType contentType;
    private byte[] byteArray;


    public ByteArrayRequestBody(MediaType contentType,byte[]byteArray){
        this.contentType=contentType;
        this.byteArray=byteArray;
    }

    @Nullable
    @Override
    public MediaType contentType() {
        return contentType;
    }

    @Override
    public long contentLength() throws IOException {
        return byteArray.length;
    }

    @Override
    public void writeTo(BufferedSink sink) throws IOException {
        sink.write(byteArray,0,byteArray.length);
    }
}

File: RxHttp/src/main/java/com/github/kongpf8848/rxhttp/ByteArrayRequestBody.kt
package com.github.kongpf8848.rxhttp

import okhttp3.MediaType
import okhttp3.RequestBody
import okio.BufferedSink
import java.io.IOException

class ByteArrayRequestBody(private val contentType: MediaType?, private val byteArray: ByteArray?) : RequestBody() {
    override fun contentType(): MediaType? {
        return contentType
    }

    @Throws(IOException::class)
    override fun contentLength(): Long {
        return byteArray!!.size.toLong()
    }

    @Throws(IOException::class)
    override fun writeTo(sink: BufferedSink) {
        sink.write(byteArray, 0, byteArray!!.size)
    }

}